datafusion/physical-expr/src/expressions/no_op.rs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//! Literal expressions for physical operations
+
+use std::any::Any;
+use std::sync::Arc;
+
+use arrow::{
+    datatypes::{DataType, Schema},
+    record_batch::RecordBatch,
+};
+
+use crate::physical_expr::down_cast_any_ref;
+use crate::PhysicalExpr;
+use datafusion_common::Result;
+use datafusion_expr::ColumnarValue;
+
+/// A place holder expressions, can not be evaluated
+#[derive(Debug, PartialEq, Eq, Default)]
+pub struct NoOp {}
+
+impl NoOp {
+    /// Create a NoOp expression
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl std::fmt::Display for NoOp {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "NoOp")
+    }
+}
+
+impl PhysicalExpr for NoOp {
+    /// Return a reference to Any that can be used for downcasting
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn data_type(&self, _input_schema: &Schema) -> Result<DataType> {
+        Ok(DataType::Null)
```

Comment:
🤔  since this acts like a `PhysicalExpr::Literal` for `ScalarValue::Null` I wonder if you considered using `PhysicalExpr::Literal` directly instead (for CaseExpr)?

Reply:
The `NoOp` expr type will also be used in other places in the following PR.